drivers/SPI.cpp

```diff
@@ -16,97 +16,146 @@
 #include "drivers/SPI.h"
 #include "platform/mbed_critical.h"
 
-#if DEVICE_SPI_ASYNCH
+#if 0
```

Comment:
Why is this code disabled?

Reply:
This is because the async API in the driver is not very clear on :
- Locks :
  - Shall lock match the lock of the peripheral (thread safety of the peripheral)
  - Shall lock match the lock of the driver instance to race (thread safety of the instance)
- chip select behaviours
  - How should that behave when chip select is handled in hardware ?
  - What is the relation between the chip select and the locks ?